src/api/jacobian.jl

```diff
@@ -43,21 +43,18 @@ function jacobian{A}(f, ::Type{A}=Void;
                      chunk_size::Int=default_chunk_size,
                      cache::ForwardDiffCache=ForwardDiffCache(),
                      output_length::Int=0)
-    # if output_length > 0, assume that f is of 
-    # the form f!(output, x), and generate the 
+    # if output_length > 0, assume that f is of
+    # the form f!(output, x), and generate the
     # appropriate closure
     if output_length > 0
-        newf = @eval begin
-            output_cache = ForwardDiffCache()
-            @inline function newf{N,T,C}(x::Vector{GradientNumber{N,T,C}})
-                output = get_workvec!(output_cache, 
-                                      GradientNumber, T,
-                                      Val{$output_length}, 
-                                      Val{N})
-                $f(output, x)
-                return output
-            end
-            return newf
+        output_cache = ForwardDiffCache()
+        @inline function newf{N,T,C}(x::Vector{GradientNumber{N,T,C}})
```

Comment:
> newf is passed in at runtime though. The function body isn't known at compile time. Same reason why passing functions isn't well optimized in julia.

I just realized that after rereading my own question above! Okay, I see what you mean now